core/sql/exp/exp_like.cpp

```diff
@@ -513,11 +512,16 @@ ex_expr::exp_return_type ExRegexpClauseChar::eval(char *op_data[],
 
   str_cpy_all(pattern, op_data[2], len2);
   str_cpy_all(srcStr, op_data[1], len1);
-
-  z = regcomp(&reg, pattern, cflags);
+  if(rpattern_ != pattern)
```

Comment:
This is a pointer comparison rather than a content comparison. Since rpattern_ is a data member and pattern is a stack variable, I expect this comparison will always mismatch, so (rpattern_ != pattern) will always be true. I am guessing that you intended to compare the content instead?

Reply:
Hi, Dave, rpattern_ is NAString, so I think this comparison is correct? My intent is to compare the content.
Let me check the NAString further.